### PR TITLE
Add CIS GCP 6.4

### PR DESCRIPTION
**Pull request by orouz**
### Summary of your changes

Adds CIS GCP 6.4 

| failed (ssl is not required) |  passed (ssl is required)  |
|-----------|---|
| Screenshot 2023-08-09 at 9 34 39 | - |

### Related Issues
 - https://github.com/elastic/csp-security-policies/issues/245

### Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

#### Introducing a new rule?
- [x] Generate rule metadata using [this script](https://github.com/elastic/csp-security-policies/tree/main/dev#generate-rules-metadata)
- [x] Add relevant unit tests
- [ ] Generate relevant rule templates using [this script](https://github.com/elastic/csp-security-policies/tree/main/dev#generate-rule-templates), and open a PR in [elastic/packages/cloud_security_posture](https://github.com/elastic/integrations/tree/main/packages/cloud_security_posture)

> **Note** Make sure to bump cloudbeat [`version/policy.go`](https://github.com/elastic/cloudbeat/blob/main/version/policy.go#L20) after merging this PR and drafting a new release.

### CODE REVIEW

File: bundle/compliance/cis_gcp/rules/cis_6_4/rule.rego [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package compliance.cis_gcp.rules.cis_6_4
+
+import data.compliance.lib.common
+import data.compliance.policy.gcp.data_adapter
+import future.keywords.if
+
+default ssl_is_required = false
```
Comment (oren-zohar) [inline bundle/compliance/cis_gcp/rules/cis_6_4/rule.rego:7]: you can use `else = false` instead

**Review by uri-weisman — COMMENTED** [review]
just wondering about:
> This recommendation is applicable for Postgresql, MySql generation 1, MySql generation 2 and SQL Server 2017 instances.

File: bundle/compliance/cis_gcp/rules/cis_6_4/rule.rego [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	# Gen 2
+	"MYSQL_8_0",
+	"POSTGRES_9_6",
+	"POSTGRES_10",
+	"POSTGRES_11",
+	"POSTGRES_12",
+	"POSTGRES_13",
+	"POSTGRES_14",
+	"POSTGRES_15",
+}
```
Comment (orouz) [inline bundle/compliance/cis_gcp/rules/cis_6_4/rule.rego:38]: > This recommendation is applicable for Postgresql, MySql generation 1, MySql generation 2 and SQL Server 2017 instances.

- all of postgresql 
- mysql gen 1 and 2 
- sqlserver 2017

Comment (orouz) [reply bundle/compliance/cis_gcp/rules/cis_6_4/rule.rego:38]: another approach:
```rego
is_relevant_sql_instance if {
	startswith(data_adapter.resource.data.databaseVersion, "POSTGRES")
} else = false

is_relevant_sql_instance if {
	data_adapter.resource.data.databaseVersion in {"MYSQL_5_6", "MYSQL_5_7", "MYSQL_8_0"}
} else = false

is_relevant_sql_instance if {
	startswith(data_adapter.resource.data.databaseVersion, "SQLSERVER_2017")
} else = false
```

wdyt?

Comment (uri-weisman) [reply bundle/compliance/cis_gcp/rules/cis_6_4/rule.rego:38]: cool, I prefer the current approach.
Will the size of a blacklist be the same?
I'm concerned about a scenario in which a new version is supported but we'll not evaluate those instances (e.g new postgres version).

Comment (orouz) [reply bundle/compliance/cis_gcp/rules/cis_6_4/rule.rego:38]: the current approach uses the current values in GCP but doesn't account for newly added values as we can't tell what those would be in advance. (like in `postgres`). the second approach does account for it though.

Comment (uri-weisman) [reply bundle/compliance/cis_gcp/rules/cis_6_4/rule.rego:38]: so let's go with the second approach then

**Review by uri-weisman — APPROVED** [review]

**Pull request merged**